CMakeLists.txt

```diff
@@ -30,11 +30,42 @@ file(GLOB_RECURSE TVL_GENERATOR_SOURCES CONFIGURE_DEPENDS
 )
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${TVL_GENERATOR_SOURCES})
 
+# check for C++20 Concepts features
+if("cxx_std_20" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+	message("Compiler supports C++20")
+	#set(CMAKE_CXX_STANDARD 20) #this seems like a bad idea
+	set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -std=c++20") #todo: standard should be set in a portable way
```

Comment:
What about using `target_compile_features(tvl INTERFACE cxx_std_20)` ?

Reply:
> target_compile_features(tvl INTERFACE cxx_std_17)

+1


> affects the current+children cmake

But this may be an issue here. If the CMakeLists.txt is included in some top-level CMakeLists.txt from another project, this muddles with the top-level project or its dependencies (depending on the position of the include), doesn't it? If not, then this should be fine. However, as far as I understood, the documentation "CMAKE_REQUIRED_FLAGS" is the preferred way.

Reply:
If another project uses `add_subdirectory` for us, our set(...) will not affect their value of CMAKE_CXX_STANDARD (subdirectory opens a new variable scope). If they use `include` (no separate variable scope), that is what they wanted, not our fault. It is standard practice to do this.